docs/book/src/reference/watching-resources/testdata/external-indexed-field/controller.go

```diff
@@ -169,7 +169,7 @@ func (r *ConfigDeploymentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		For(&appsv1.ConfigDeployment{}).
 		Owns(&kapps.Deployment{}).
 		Watches(
-			&source.Kind{Type: &corev1.ConfigMap{}},
+			&corev1.ConfigMap{},
 			handler.EnqueueRequestsFromMapFunc(r.findObjectsForConfigMap),
 			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
 		).
```

Comment:
The syntax shows that changed, see: how it should be: https://github.com/kubernetes-sigs/controller-runtime/blob/1f5b39fa59d15fae78e521c9c9f2acabbbb3ea17/pkg/handler/example_test.go#L54-L63

Can you please try to use it as described in the above example?
And let us know if worked for you